kube-core/src/discovery.rs

```diff
@@ -57,6 +58,19 @@ impl ApiResource {
     }
 }
 
+impl Into<APIResource> for ApiResource {
+    fn into(self) -> APIResource {
+        APIResource {
+            group: Option::from(self.group),
+            kind: self.kind,
+            version: Option::from(self.api_version),
```

Comment:
These first 3 are correct. Although I'd use `Some(self.group)` etc.

You are missing `name` which comes from `self.plural`.

Reply:
we could `Some` but what if it had empty string?

Reply:
empty string is a legal group name